drivers/serial/uart_ns16550.c

```diff
@@ -293,6 +293,11 @@ static void set_baud_rate(struct device *dev, u32_t baud_rate)
 		 */
 		divisor = ((dev_cfg->sys_clk_freq + (baud_rate << 3))
 					/ baud_rate) >> 4;
+#if defined(CONFIG_SOC_SERIES_CEC)
```

Comment:
This needs to be more abstracted.  We should introduce an additional DTS compatible for the uart on the SoC Series and use the `DT_COMPAT_<FOO>` define to base this ifdef on.  Also we need to define the magic number

Reply:
Updated per request. I appended the additional compatible string to the list so the normal config options will work as-is. Invented some symbolic names (based on CEC data sheet) for the flag, and used the system clock config as proper. Is something more needed?